docs/pages/guides/spend-permissions/api-reference.mdx

```diff
@@ -0,0 +1,276 @@
+
+```solidity
+function revoke(SpendPermission calldata spendPermission) external requireSender(spendPermission.account);
+```
+
+---
+
+#### `getHash`
+
+Generate a hash of a `SpendPermission` struct for signing, in accordance with [EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md).
+
+```solidity
+function getHash(SpendPermission memory spendPermission) public view returns (bytes32);
+```
+
+---
+
+#### `getBatchHash`
+
+Generate a hash of a `SpendPermissionBatch` struct for signing, in accordance with [EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md). 
+Reverts if the batch is empty.
+
+```solidity
+function getBatchHash(SpendPermissionBatch memory spendPermissionBatch) public view returns (bytes32);
+```
+
+---
+
+#### `isApproved`
+
+Check if a spend permission is approved and not revoked.
+
+```solidity
+function isApproved(SpendPermission memory spendPermission) public view returns (bool);
+```
+
+---
+
+#### `getCurrentPeriod`
+
+Retrieve the start, end, and accumulated spend for the current period of a spend permission.
+
+```solidity
+function getCurrentPeriod(SpendPermission memory spendPermission) public view returns (PeriodSpend memory);
+```
+## Coinbase Smart Wallet clients
+
+### Using Viem
+[Viem](https://viem.sh/) is a TypeScript interface for Ethereum that provides low-level stateless primitives for interacting with Ethereum.
+
+- **[Getting Started with Viem](https://viem.sh/docs/getting-started.html)** 
+
+- **[Getting Started with Account Abstraction](https://viem.sh/account-abstraction#getting-started-with-account-abstraction)** 
+   Viem provides support for account abstraction clients, including specific support for Coinbase Smart Wallet and paymasters.
+
+### Using OnchainKit
+[OnchainKit](https://onchainkit.xyz/) is a collection of React components and TypeScript utilities that help developers quickly build
+onchain applications.
+
+### Using Wagmi
+[Wagmi](https://wagmi.sh/) is a collection of React Hooks that facilitate development of blockchain frontends.
+
+-  **[Getting Started with Wagmi](https://wagmi.sh/react/getting-started)** 
+- **[coinbaseWallet](https://wagmi.sh/react/api/connectors/coinbaseWallet)** 
+
+
+## Coinbase Wallet API
+
+### `wallet_fetchPermissions`
+
+#### Schema
+```typescript
+type FetchPermissionsRequest = {
+  account: Address;
+  chainId: string; // 0x...
+  spender: Address;
+  pageOptions: PageOptions; // optional, defaults to page size 50 and empty cursor
+}
+
+type FetchPermissionsResponse = {
+  permissions: FetchPermissionsResponseItem[];
+  pageDescription: PageDescription;
+}
+
+type PageOptions = {
+  pageSize number; // number of items requested, defaults to 50
+  cursor string; // identifier for where the page should start
+}
+
+type PageDescription = {
+  pageSize number; // number of items returned
+  nextCursor string; // identifier for where the next page should start
+}
```

Comment:
I get why these are nested into their own objects, but it just adds a bit more clutter for someone trying to see the whole shape of the API to have separate types. Can we remove these types and have the inner object in the other types, e.g.
```typescript
type FetchPermissionsResult = {
  permissions: FetchPermissionsResultItem[];
  pageDescription: {
    pageSize: number;
    nextCursor: string;
  }
}